doc/dev/tests.md

```diff
@@ -0,0 +1,378 @@
+* Creation and cleanup of helper resources, such as resource groups, storage accounts, etc. which can be used in order to test services
+* Processors for modifying requests and responses when writing or reading recordings (for example, to to avoid recording credential information)
+* Patches for overriding functions and methods that don't work well with testing frameworks (such as long-running operations)
+
+Code in the [`azure-sdk-tools/devtools_testutils`](https://github.com/Azure/azure-sdk-for-python/tree/master/tools/azure-sdk-tools/devtools_testutils) directory provides concrete implementations of the features provided in `scenario_tests` that are oriented around use in SDK testing and that you can use directly in your unit tests.
+
+## Define credentials
+When you run tests in playback mode, they use a fake credentials file, located at `tools/azure-sdk-tools/devtools_testutils/mgmt_settings_fake.py` to simulate authenticating with Azure. The test framework uses the values stored in the `mgmt_settings_real.py` to take care of authenticating clients for you.
+
+In live mode, the credentials need to be real so that the tests are able to connect to the service. Copy the `mgmt_settings_fake.py` file to a new file named `mgmt_settings_real.py` within the same directory. Then make the following changes:
+1. Change the value of the `SUBSCRIPTION_ID` variable to your organizations subscription ID. If you don't have it, you can find it in the "Overview" section of the "Subscriptions" blade in the [Azure portal](https://portal.azure.com/).
+2. Create a `.env` file at the root of the repository (in the same directory as the `sdk`, `tools`, `eng` folders). In this file you can define any environment variables you need for a test and that will be loaded by the `AzureTestCase` file. Start by defining `AZURE_TENANT_ID`, `AZURE_CLIENT_ID`, and `AZURE_CLIENT_SECRET` which are available after creating a Service Principal or can be retrieved from the Azure Portal if you have already created a Service Principal. This file stores secrets in plain text so it is important that the contents of this file are not committed to the git repository.
+3. Change the `get_azure_core_credentials(**kwargs):` function to construct and return a `ClientSecretCredential` object. The `client_id`, `client_secret`, `tenant_id` are provided when you create a service principal. These values can be found in the Azure Portal. This method in your `mgmt_settings_real.py` file should look like this:
+```python
+def get_azure_core_credentials(**kwargs):
+    from azure.identity import ClientSecretCredential
+    import os
+    return ClientSecretCredential(
+        client_id = os.environ['AZURE_CLIENT_ID'],
+        client_secret = os.environ['AZURE_CLIENT_SECRET'],
+        tenant_id = os.environ['AZURE_TENANT_ID']
+    )
+```
+4. In the same directory as your `mgmt_settings_real.py` file create a `testsettings_local.cfg` and copy and paste the following line:
```

Comment:
I prefer your paragraph you wrote elsewhere about env variable. We could keep that sentence, but then let's mention the right folder directly (and not "where mgmt-settings_real is")